### DISCUSSION TITLE
Serve Assets from CDN results in CORS errors

### DISCUSSION THREAD

**Discussion by greimers** [livewire/livewire#3108] [7. Bugs (archived)]
### Description
I thought it would be elegant to serve the livewire assets JS over my CDN. 
So I set the `asset_url` in the config to `https://cdn.mydomain.com`. This works fine regarding the loading of the JS files. The CDN will grab the latest version from the server and then serve this to the browser.

But when I get CORS errors in the console as soon as I trigger a LiveWire action because LiveWire now sends POST XHR requests also to `https://cdn.mydomain.com/livewire/message/something` instead of `https://mydomain.com/livewire/message/something`.

Am I totally misunderstanding something or is this a bug? I'm not deep into the internals of how LiveWire does its magic.
I would naively expect LiveWire to use the `APP_URL` for the XHR requests and not the asset URL. 

**Context**
- Livewire version: v2.3.17
- Laravel version: 8.0
- Browser: Firefox

**Comment by hectordommo** [comment]
I have a similar problem, where Im serving my app over https, but all "post" request trying to upload files are send via http, so I can't upload files. Wonder if problems are similars. Have you find a workaround your problem ?

I get this error:

`Mixed Content: The page at 'https://domain.mx/staff/project/credit/c3ade6f6-6ec5-4a8d-ab3b-4dd910486a14' was loaded over HTTPS, but requested an insecure XMLHttpRequest endpoint 'http://domain.mx/livewire/upload-file?expires=1616259115&signature=6c71f7b41a6383f27ab94e60b6a40f3cb6ff6e62b2630b7c3bd295810273d208'. This request has been blocked; the content must be served over HTTPS.`

Livewire 2.4
Laravel 8.33
Chrome bowser 89.0.438

**Comment by greimers** [comment]
@hectordommo  No. I'm now serving the LiveWire assets from the default path (so without CDN). 
I haven't tried uploads with LiveWire yet. So unfortunately I can't give you any hint for your HTTP/HTTPS issue. 
Have you tried resetting any LiveWire config modifications you might have made?

**Comment by hectordommo** [comment]
@greimers Thank you for your response. I did clear all configuration, and caches. I think some new version mess up my installation, since every time I deploy everything gets reinstalled. Before last week It was working. I'll try to figure out what happend and report back.

**Comment by jskorlol** [comment]
@hectordommo same problem...
Call upload-file to http.. How can I solve this?

Somehow, this seems to be a problem of complex configuration elements... On the laravel side, not livewire?
I'm using treafik, and there seems to be a problem with any configuration used.

So the way I solved
https://stackoverflow.com/questions/47100382/urlforceschema-not-working-before-login/47341952#47341952

So, if you print the route, https is called unconditionally. The upload of livewire is also good.
It would be better if I solve the exact cause, but...

**Comment by mityukov** [comment]
Same issue here. I wanted to unload static files to another server, so I added the following to my .env:
```env
ASSET_URL=https://static.mysite.com
MIX_ASSET_URL=${ASSETS_URL}
```

Then, in config/livewire.php, I'm refrencing one of these vars:
```php
'asset_url'  => env('ASSET_URL', null),
```

Now it loads JS file from https://static.mysite.com/vendor/livewire/livewire.js?id=123412341234, which is totally expected.
But it also sends XHR request to that static serving site: 
https://static.mysite.com/livewire/message/supa-dupa-component-name -- which is completely unexpected 😭 

laravel 8 & livewire/livewire v2.12.8

**Reply by lrljoe** [reply]
@mityukov 
I'd strongly urge you to upgrade to a supported version of Laravel, and Livewire.  Shift can help you here.

Your config/livewire.php should reference the path for your site, rather than your static assets

**Reply by mityukov** [reply]
> Your config/livewire.php should reference the path for your site, rather than your static assets

Got it. So, is there a way to tell Livewire where to grab specifically _assets_ (that livewire JS file) from?
